core/services/keeper/registry_synchronizer_core.go

```diff
@@ -75,12 +82,16 @@ func (rs *RegistrySynchronizer) Start(context.Context) error {
 
 		var upkeepPerformedFilter [][]log.Topic
 		upkeepPerformedFilter = nil
+		topicHash := rs.job.KeeperSpec.FromAddress.Hash()
+		if rs.forwardingAllowed && (rs.forwarderAddress != common.Address{}) {
```

Comment:
can we add the `(rs.forwarderAddress != common.Address{})` somewhere upstream so that it doesn't need to be checked everytime

Reply:
I see the logic to override fromAddress multiple times. Maybe we can simplify this by having this logic in the delegate which passes only two addresses

- eoaAddress: only used in txm pipeline argument
- effectiveFromAddress: result of the combination of from/forwarder logic

Reply:
Agreed. changed logic to only do the check once in the delegate. effectiveKeeperAddress is the keeper on the registry, fromAddress is EOA. I'm reluctant to change fromAddress to eoaAddress as I want fromAddress to ALWAYS mean eoa for us